src/gc-stacks.c

```diff
@@ -215,33 +216,38 @@ void sweep_stack_pools(void) JL_NOTSAFEPOINT
     //            bufsz = t->bufsz
     //            if (stkbuf)
     //                push(free_stacks[sz], stkbuf)
-    assert(gc_n_threads);
```

Comment:
Why was this assertion removed?

Reply:
Because this code can now be accessed by a thread that can enter this function after the GC is done. It will just exit but it means this assertion isn't always true anymore. I moved the assertion to the place that calls this. Another option is to sink the assert to inside the loop. Not sure what you think